core/src/main/java/org/apache/calcite/sql/SqlTableFunction.java

```diff
@@ -30,4 +32,15 @@ public interface SqlTableFunction {
    * @return strategy to infer the row type of a call to this function
    */
   SqlReturnTypeInference getRowTypeInference();
+
+  /**
+   * Returns the table parameter characteristics for <code>ordinal</code>th argument to this
+   * table function.
```

Comment:
argument or parameter? I think probably parameter.

does it return Optional.empty? Or does it return null?

what is the behavior if ordinal is < 0 or >= the number of parameters?

Reply:
Good catch.
It would return null if the ordinal Nth argument is not table parameters or the original  is < 0 or >= the number of parameters.
I've updated the comments of the function.